repositories/repo_repository.py

```diff
@@ -1,6 +1,42 @@
-import sqlalchemy as sa
+import logging
 
+from database import (
+    get_connection, repository_t
+)
 
-class RepoRepository:
+from .abstract_repository import AbstractRepository
 
-    cache = {}
+log = logging.getLogger('application')
+
+
+class RepoRepository(AbstractRepository):
+
+    def __init__(self):
+        self.cache = {}
+        super().__init__(repository_t)
+
+    def find(self, **kwargs):
+        cache_key = self.get_cache_key_from_kwargs(prefix='find', **kwargs)
+        if cache_key in self.cache:
+            self.history.add_cached()
+            return self.cache[cache_key]
+        with get_connection() as conn:
+            whereclause = self.build_whereclause_from_kwargs(**kwargs)
+            users = conn.execute(self.tbl.select().where(whereclause=whereclause)).fetchall()
+            if users:
+                self.cache.setdefault(cache_key, users)
+                self.history.add_successful()
+            return users
+
+    def find_one(self, **kwargs):
```

Comment:
I think arg of find_one should be more predicable, some thing like pk in Django models